manifests/osfamily/debian.pp

```diff
@@ -7,6 +7,14 @@
     key        => {
       'id'     => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
       'server' => 'pgp.mit.edu',
-    }
+    },
+    notify     => Notify['pc1_repo_force'],
+  }
+
+  # apt_update doesn't inherit the future class dependency, so it
+  # can wait until the end of the run to exec. Force it to happen now.
+  notify { 'pc1_repo_force':
```

Comment:
with this PR I see the refresh before we try to install puppet-agent (as required):

```
Notice: /Stage[main]/Apt::Update/Exec[apt_update]: Triggered 'refresh' from 1 events
Notice: forcing apt update for pc1_repo
Notice: /Stage[main]/Puppet_agent::Osfamily::Debian/Notify[pc1_repo_force]/message: defined 'message' as 'forcing apt update for pc1_repo'
```

It's interesting that the sentinel is required. It's as though the refresh is not contained within this class, but adding a sentinel forces it. Seems like it might be a puppet bug?